internal/ethapi/api.go

```diff
@@ -2644,6 +2645,45 @@ func checkTxFee(gasPrice *big.Int, gas uint64, cap float64) error {
 }
 
 // CheckTxFee exports a helper function used to check whether the fee is reasonable
-func CheckTxFee(gasPrice *big.Int, gas uint64, cap float64) error {
-	return checkTxFee(gasPrice, gas, cap)
+func CheckTxFee(ctx context.Context, backend CeloBackend, gasPrice *big.Int, gas uint64, feeCurrency *common.Address) error {
+	feeCap, err := ConvertTxFeeCapToCurrency(ctx, backend, feeCurrency)
```

Comment:
Why are you converting the multiplier rather than the gas-price? 

You could write a wrapper function that takes in the transaction and Celo-backend, and returns the celo-denominated gas-price.
Then you pass in this price into the unmodified `checkTxFee(gasPrice, gas, cap)` function.
Like this you wouldn't have the rational number conversion and could reuse the existing conversion functions

Reply:
@ezdac Yes, right. It will be better one. I refactored CheckTxFee and totally changed tests. So resolved other current comments. Please check the latest codes. Thank you.